hash/src/lib.rs

```diff
@@ -19,7 +19,7 @@ use {
     },
     solana_sanitize::Sanitize,
 };
-#[cfg(target_arch = "wasm32")]
+#[cfg(feature = "js")]
```

Comment:
Assuming that we gate std just on the std feature, let's gate this properly with:

```
#[cfg(all(feature = "std", feature = "js"))]
```

For reference, I'm looking at how uuid declares something similar here: https://github.com/uuid-rs/uuid/blob/cb19a46cf17cd9840205e0d67a32ea8a18e58374/src/timestamp.rs#L308

I don't think we need the full target_arch part too, but the std + js feature gate seems to make sense